pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -216,3 +215,25 @@ func (m *OSDHealthMonitor) updateCephStatus(devices []string) {
 		}
 	}
 }
+
+func (m *OSDHealthMonitor) getOSDStoreStatus() (*cephv1.OSDStatus, error) {
+	label := fmt.Sprintf("%s=%s", k8sutil.AppAttr, AppName)
+	osdDeployments, err := k8sutil.GetDeployments(m.clusterInfo.Context, m.context.Clientset, m.clusterInfo.Namespace, label)
```

Comment:
The OSD deployments will rarely change, so it seems expensive to query these deployments again every minute for the OSD status. I wonder if we can skip this query most of the time. For example, perhaps only query if the desired storeType is different from any existing OSD store types.

Reply:
Or perhaps we just query this and update the status at the end of a normal OSD reconcile. During the OSD status update just seems too frequent.

Reply:
>> Or perhaps we just query this and update the status at the end of a normal OSD reconcile. During the OSD status update just seems too frequent.

With this approach, we will updating `osdStatus` at two different places (one for deviceClass list and one for OSD store).  That might be more confusing to debug later on.

Reply:
>> The OSD deployments will rarely change, so it seems expensive to query these deployments again every minute for the OSD status. I wonder if we can skip this query most of the time. For example, perhaps only query if the desired storeType is different from any existing OSD store types.

I prefer this approach. We just need to pass the `desired store type` spec to the OSD health checker.  If  desired store type is `bluestore-rdr` but `status.storage.osd.storeType['bluestore`] != 0` (implying that migration is not complete), only then we make go ahead and get the OSD list.

Reply:
Agreed, that should work perfectly to reduce the queries, and keep the osd status update in a single place.

Reply:
The approach I suggested will not work.

Reply:
I'll take this up in a follow PR next week.